src/main/java/org/folio/entlinks/service/links/AuthorityDataStatService.java

```diff
@@ -34,4 +47,110 @@ public List<AuthorityDataStat> createInBatch(List<AuthorityDataStat> stats) {
+    log.info("Updating links, stats for reports: [jobId: {}, reports count: {}]", jobId, reports.size());
+    log.debug("Updating links,stats for reports: [reports: {}]", reports);
+    updateLinks(reports);
+    updateStatsData(jobId, reports);
+  }
+
+  private void updateLinks(List<LinkUpdateReport> reports) {
+    reports.forEach(report -> {
+      var linkIds = report.getLinkIds();
+      var links = linkingService.getLinksByIds(linkIds);
+
+      links.forEach(link -> {
+        link.setStatus(mapReportStatus(report));
+        link.setUpdatedAt(now());
+        if (report.getStatus().equals(FAIL)) {
+          link.setErrorCause(report.getFailCause());
+        } else {
+          link.setErrorCause(EMPTY);
+        }
+      });
+
+      linkingService.saveAll(report.getInstanceId().toString(), links);
+    });
+  }
+
+  private InstanceAuthorityLinkStatus mapReportStatus(LinkUpdateReport report) {
+    switch (report.getStatus()) {
+      case SUCCESS -> {
+        return InstanceAuthorityLinkStatus.ACTUAL;
+      }
+      case FAIL -> {
+        return InstanceAuthorityLinkStatus.ERROR;
+      }
+      default -> throw new IllegalArgumentException("Unknown link update report status.");
+    }
+  }
+
+  /**
+   * Updates authority statistics data.
+
+   * @param jobId linked bib update job id.
+   *              AuthorityDataStat id and jobId are interchangeable (jobId is used as id to create stat record)
+   * */
+  private void updateStatsData(UUID jobId, List<LinkUpdateReport> reports) {
+    var dataStat = statRepository.findById(jobId)
+      .orElseThrow(() -> new IllegalStateException("Cannot find authority data statistics for id: " + jobId));
+    var failedCount = getReportCountForStatus(reports, FAIL);
+    var successCount = getReportCountForStatus(reports, SUCCESS);
+
+    if (failedCount != 0) {
+      dataStat.setFailCause(getStatFailCause(dataStat, reports));
+    }
+    dataStat.setLbUpdated(dataStat.getLbUpdated() + successCount);
+    dataStat.setLbFailed(dataStat.getLbFailed() + failedCount);
+
+    var jobCompleted = dataStat.getLbUpdated() + dataStat.getLbFailed() == dataStat.getLbTotal();
+    if (jobCompleted) {
+      dataStat.setCompletedAt(now());
+      updateStatStatus(dataStat);
+    }
+
+    log.info("Saving stats data [statsId: {}, status: {}]", dataStat.getId(), dataStat.getStatus());
+    log.debug("Stats data: {}", dataStat);
+    statRepository.save(dataStat);
+  }
+
+  private int getReportCountForStatus(List<LinkUpdateReport> reports, LinkUpdateReport.StatusEnum status) {
+    return (int) reports.stream()
+      .filter(linkUpdateReport -> linkUpdateReport.getStatus().equals(status))
+      .count();
+  }
+
+  private String getStatFailCause(AuthorityDataStat dataStat, List<LinkUpdateReport> reports) {
+    var newFailCause = reports.stream()
+      .map(LinkUpdateReport::getFailCause)
+      .filter(StringUtils::isNotBlank)
+      .collect(Collectors.joining(FAIL_CAUSE_DELIMITER));
```

Comment:
What do we have in the database? Will it be enough to save all errors, what if we will have 1000 errors?